### PR TITLE
feat(notifications): Team Notification Settings

### PR DESCRIPTION
**Pull request by mgaeta**
This PR creates the Team Notification Settings Page to [match this spec](https://www.figma.com/file/48ATxgADirZkRz3qkV46e0/Handover%3A-Notifications?node-id=288%3A2425).
Screen Shot 2021-07-08 at 2 54 30 PM
Screen Shot 2021-07-08 at 3 05 39 PM
Its only job is to show a teams linked external identities, not add, update, or delete them.

### CODE REVIEW

File: static/app/views/settings/organizationTeams/teamNotifications.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    const externalTeams = teamDetails.externalTeams.filter(externalTeam =>
+      NOTIFICATION_PROVIDERS.includes(externalTeam.provider)
+    );
+
+    if (!externalTeams.length) {
+      return (
+        <EmptyMessage>
+          <div>{t('No External Teams have been linked yet.')}</div>
+          <NotDisabledSubText>
+            Head over to Slack and type <code>/sentry</code> to get started.{' '}
```
Comment (ceorourke) [inline static/app/views/settings/organizationTeams/teamNotifications.tsx:75]: shouldn't this instruct them to either type `sentry help` or `sentry link team`?

Comment (mgaeta) [reply static/app/views/settings/organizationTeams/teamNotifications.tsx:75]: I'll update it to say `/sentry link team`

File: src/sentry/api/endpoints/team_details.py [hunk 2]
```diff
@@ -40,9 +42,14 @@ def get(self, request, team):
         :pparam string organization_slug: the slug of the organization the
                                           team belongs to.
         :pparam string team_slug: the slug of the team to get.
+        :qparam bool full: if this is set to true then the team object will
+            include projects and external_teams. Set to 1 to enable.
         :auth: required
         """
-        context = serialize(team, request.user)
+        full = request.GET.get("full", False)
```
Comment (mgaeta) [inline src/sentry/api/endpoints/team_details.py:49]: I'm going to do `expand=["projects"]` here instead.

File: static/app/views/settings/organizationTeams/teamNotifications.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+class TeamNotificationSettings extends AsyncView<Props, State> {
+  getTitle() {
+    return 'Team Notification Settings';
+  }
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {organization, team} = this.props;
+    return [
+      ['teamDetails', `/teams/${organization.slug}/${team.slug}/`, {query: {full: true}}],
```
Comment (NisanthanNanthakumar) [inline static/app/views/settings/organizationTeams/teamNotifications.tsx:34]: @mgaeta you will need to update this query if moving towards using `expand`

File: static/app/views/settings/organizationTeams/teamNotifications.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    const externalTeams = teamDetails.externalTeams.filter(externalTeam =>
+      NOTIFICATION_PROVIDERS.includes(externalTeam.provider)
+    );
+
+    if (!externalTeams.length) {
+      return (
+        <EmptyMessage>
+          <div>{t('No External Teams have been linked yet.')}</div>
+          <NotDisabledSubText>
+            Head over to Slack and type <code>/sentry link team</code> to get started.{' '}
```
Comment (NisanthanNanthakumar) [inline static/app/views/settings/organizationTeams/teamNotifications.tsx:75]: @mgaeta we can use the `tct` utility function to translate this

Comment (NisanthanNanthakumar) [reply static/app/views/settings/organizationTeams/teamNotifications.tsx:75]:
```suggestion
tct('Head over to Slack and type [code] to get started. [link].', {
            link: (
              <Link>{t('Learn more')}</Link>
            ),
            code: <code>/sentry link team</code>
          })
```

File: static/app/views/settings/organizationTeams/teamNotifications.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        disabled
+        key={externalTeam.id}
+        label={
+          <div>
+            <NotDisabledText>
+              {toTitleCase(externalTeam.provider)}:
+              {integrationsById[externalTeam.integrationId].name}
+            </NotDisabledText>
+            <NotDisabledSubText>
+              Unlink this channel in Slack with <code>/sentry unlink team</code>.{' '}
```
Comment (NisanthanNanthakumar) [inline static/app/views/settings/organizationTeams/teamNotifications.tsx:97]: @mgaeta we can use `tct` here as well

File: static/app/views/settings/organizationTeams/teamNotifications.tsx (hunk 4 above)
Comment (NisanthanNanthakumar) [reply static/app/views/settings/organizationTeams/teamNotifications.tsx:75]: @mgaeta I don't know if we should translate the command. Will our slack app accept different languages of the English version?

Comment (mgaeta) [reply static/app/views/settings/organizationTeams/teamNotifications.tsx]: Let's assume that we'll accept other languages later.

File: src/sentry/api/endpoints/team_details.py [hunk 6]
```diff
@@ -40,12 +40,22 @@ def get(self, request, team):
...
         """
-        context = serialize(team, request.user)
-        context["organization"] = serialize(team.organization, request.user)
+        collapse = request.GET.getlist("collapse", [])
+        expand = request.GET.getlist("expand", [])
 
-        return Response(context)
+        # A little hack to preserve existing behavior.
+        if "organization" in collapse:
+            collapse.remove("organization")
```
Comment (NisanthanNanthakumar) [inline src/sentry/api/endpoints/team_details.py:54]: @mgaeta why have `collapse` if we're going to keep it as an empty array? I don't see it being used in the serializer.

File: src/sentry/api/serializers/models/team.py [hunk 7]
```diff
@@ -149,55 +194,24 @@ def serialize(
...
-    ) -> MutableMapping[Team, MutableMapping[str, Any]]:
-        project_teams = list(
-            ProjectTeam.objects.filter(team__in=item_list, project__status=ProjectStatus.VISIBLE)
-            .order_by("project__name", "project__slug")
-            .select_related("project")
-        )
-
-        actor_mapping = {team.actor_id: team for team in item_list}
-        external_actors = list(ExternalActor.objects.filter(actor_id__in=actor_mapping.keys()))
+        if self._expand("organization"):
```
Comment (mgaeta) [inline src/sentry/api/serializers/models/team.py:201]: @NisanthanNanthakumar `organization` expand is hidden in the diff.

Comment (NisanthanNanthakumar) [reply src/sentry/api/serializers/models/team.py:201]: @mgaeta ahh I see, so do we need the `_collapse` helper method?

Comment (NisanthanNanthakumar) [reply src/sentry/api/serializers/models/team.py:201]: since it looks like `self.collapse` will always be an empty list once it gets to the serializer

**Review by NisanthanNanthakumar — APPROVED** [review]
Nit: to remove the `self.collapse` in the serializer but I can see the case of it being used in the future.

**Pull request merged**